src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -386,17 +384,36 @@ private AddInterfaceInteraction createAddInterfaceInteractionAction(@Nonnull Pay
         CompletionStage<PaymentHandleResponse> paymentHandleResponseStage = ctpPaymentIdOptStage
                 .thenCompose(ctpPaymentIdOpt ->
                         ctpPaymentIdOpt.map(ctpPaymentId -> {
-                            AddInterfaceInteraction action = createAddInterfaceInteractionAction(restException.getDetails(), RESPONSE);
-                            return ctpFacade.getPaymentService().updatePayment(ctpPaymentId, Collections.singletonList(action))
-                                    .thenApply(ignore -> PaymentHandleResponse.ofHttpStatusAndErrorMessage(HttpStatus.valueOf(restException.getResponsecode()),
-                                            format("%s=[%s] can't be processed, details: [%s]", paymentIdType, paymentId, restException.getMessage())));
+                            return updatePaymentResponse(paymentId, paymentIdType, throwable, ctpPaymentId);
                         }).orElseGet(() -> CompletableFuture.completedFuture(
                                 PaymentHandleResponse.of404NotFound(format("%s=[%s] is not found.", paymentIdType, paymentId))
                         ))
                 );
         return paymentHandleResponseStage;
     }
 
+    private CompletionStage<PaymentHandleResponse> updatePaymentResponse(@Nullable String paymentId, @Nullable String paymentIdType,
+                                                                         @Nonnull Throwable throwable, String ctpPaymentId) {
+
+        if (throwable instanceof PaypalPlusServiceException) {
+            PayPalRESTException restException = ((PaypalPlusServiceException) throwable).getCause();
+
+            AddInterfaceInteraction action = createAddInterfaceInteractionAction(restException.getDetails(), RESPONSE);
+            return ctpFacade.getPaymentService().updatePayment(ctpPaymentId, Collections.singletonList(action))
+                    .thenApply(ignore -> PaymentHandleResponse.ofHttpStatusAndErrorMessage(HttpStatus.valueOf(restException.getResponsecode()),
+                            format("%s=[%s] can't be processed, details: [%s]", paymentIdType, paymentId, restException.getMessage())));
+        } else if(throwable instanceof IntegrationServiceException){
+            IntegrationServiceException serviceException = (IntegrationServiceException) throwable;
+
+            AddInterfaceInteraction action = createAddInterfaceInteractionAction(serviceException.getDetails(), RESPONSE);
```

Comment:
This code and the one above are quite the same. Pls make a new function and reuse it.

Reply:
886669bfd5da5e484eb784d25a69564315fe6677